packages/ckeditor5-table/src/tableutils.js

```
@@ -329,8 +348,11 @@ export default class TableUtils extends Plugin {
 		const model = this.editor.model;
 
 		const rowsToRemove = options.rows || 1;
+		const rowCount = this.getRows( table );
 		const first = options.at;
-		const last = first + rowsToRemove - 1;
+
+		// Make sure we are not removing too much, e.g. non-row elements at the end of the table.
+		const last = Math.min( first + rowsToRemove - 1, rowCount - 1 );
```

We should throw an error in case of invalid options.